Source/WebCore/platform/text/WritingMode.h

```diff
@@ -75,54 +75,44 @@ constexpr inline BlockFlowDirection writingModeToBlockFlowDirection(WritingMode
     return BlockFlowDirection::TopToBottom;
 }
 
-constexpr unsigned makeTextFlowInitalizer(BlockFlowDirection blockFlow, TextDirection direction)
-{
-    return static_cast<unsigned>(blockFlow) << 1 | static_cast<unsigned>(direction);
-}
-
 // Define the text flow in terms of the writing mode and the text direction. The first
 // part is the block flow direction and the second part is the inline base direction.
-enum TextFlow {
-    InlineEastBlockSouth = makeTextFlowInitalizer(BlockFlowDirection::TopToBottom, TextDirection::LTR),
-    InlineWestBlockSouth = makeTextFlowInitalizer(BlockFlowDirection::TopToBottom, TextDirection::RTL),
-    InlineEastBlockNorth = makeTextFlowInitalizer(BlockFlowDirection::BottomToTop, TextDirection::LTR),
-    InlineWestBlockNorth = makeTextFlowInitalizer(BlockFlowDirection::BottomToTop, TextDirection::RTL),
-    InlineSouthBlockEast = makeTextFlowInitalizer(BlockFlowDirection::LeftToRight, TextDirection::LTR),
-    InlineSouthBlockWest = makeTextFlowInitalizer(BlockFlowDirection::LeftToRight, TextDirection::RTL),
-    InlineNorthBlockEast = makeTextFlowInitalizer(BlockFlowDirection::RightToLeft, TextDirection::LTR),
-    InlineNorthBlockWest = makeTextFlowInitalizer(BlockFlowDirection::RightToLeft, TextDirection::RTL)
+struct TextFlow {
+    BlockFlowDirection blockFlowDirection;
+    TextDirection inlineBaseDirection;
```

Comment:
You need to incorporate `text-orientation` as well. See https://searchfox.org/mozilla-central/source/layout/generic/WritingModes.h#133

Reply:
I think that would make sense if we started using it, but that's not the case atm, so I'll refrain from doing that.

Reply:
I guess we need to fix https://bugs.webkit.org/show_bug.cgi?id=164507 first anyway.